geth/api/backend.go

```diff
@@ -208,14 +208,46 @@ func (b *StatusBackend) SendTransaction(ctx context.Context, args common.SendTxA
 	return rst.Hash, nil
 }
 
+func (b *StatusBackend) getVerifiedAccount(password string) (*account.SelectedExtKey, error) {
+	selectedAccount, err := b.accountManager.SelectedAccount()
+	if err != nil {
+		b.log.Warn("failed to get a selected account", "err", err)
+		return nil, err
+	}
+	// make sure that only account which created the tx can complete it
+	config, err := b.NodeManager().NodeConfig()
+	if err != nil {
+		return selectedAccount, err
```

Comment:
I am not sure we should return selected account if the verification was not passed.